sdk/communication/AzureCommunicationChat/Source/Signaling/events/chat.swift

```diff
@@ -0,0 +1,357 @@
+    ) {
+        self.properties = properties
+        self.updatedOn = updatedOn
+        self.updatedBy = updatedBy
+        super.init(threadId: threadId, version: version)
+    }
+}
+
+public class ChatThreadDeletedEvent: BaseChatThreadEvent {
+    public var deletedOn: Iso8601Date?
+    public var deletedBy: SignalingChatParticipant?
+
+    init(
+        threadId: String,
+        version: String,
+        deletedOn: Iso8601Date?,
+        deletedBy: SignalingChatParticipant?
+    ) {
+        self.deletedOn = deletedOn
+        self.deletedBy = deletedBy
+        super.init(threadId: threadId, version: version)
+    }
+}
+
+public class ParticipantsAddedEvent: BaseChatThreadEvent {
+    public var addedOn: Iso8601Date?
+    public var participantsAdded: [SignalingChatParticipant]?
+    public var addedBy: SignalingChatParticipant?
+
+    init(
+        threadId: String,
+        version: String,
+        addedOn: Iso8601Date?,
+        participantsAdded: [SignalingChatParticipant]?,
+        addedBy: SignalingChatParticipant?
+    ) {
+        self.addedOn = addedOn
+        self.participantsAdded = participantsAdded
+        self.addedBy = addedBy
+        super.init(threadId: threadId, version: version)
+    }
+}
+
+public class ParticipantsRemovedEvent: BaseChatThreadEvent {
+    public var removedOn: Iso8601Date?
+    public var participantsRemoved: [SignalingChatParticipant]?
+    public var removedBy: SignalingChatParticipant?
+
+    init(
+        threadId: String,
+        version: String,
+        removedOn: Iso8601Date?,
+        participantsRemoved: [SignalingChatParticipant]?,
+        removedBy: SignalingChatParticipant?
+    ) {
+        self.removedOn = removedOn
+        self.participantsRemoved = participantsRemoved
+        self.removedBy = removedBy
+        super.init(threadId: threadId, version: version)
+    }
+}
+
+public enum ChatEventId: String {
+    case chatMessageReceived
+    case typingIndicatorReceived
+    case readReceiptReceived
+    case chatMessageEdited
+    case chatMessageDeleted
+    case chatThreadCreated
+    case chatThreadPropertiesUpdated
+    case chatThreadDeleted
+    case participantsAdded
+    case participantsRemoved
+
+    init(forCode code: Int) throws {
+        switch code {
+        case 200:
+            self = .chatMessageReceived
+        case 245:
+            self = .typingIndicatorReceived
+        case 246:
+            self = .readReceiptReceived
+        case 247:
+            self = .chatMessageEdited
+        case 248:
+            self = .chatMessageDeleted
+        case 257:
+            self = .chatThreadCreated
+        case 258:
+            self = .chatThreadPropertiesUpdated
+        case 259:
+            self = .chatMessageDeleted
+        case 260:
+            self = .participantsAdded
+        case 261:
+            self = .participantsRemoved
+        default:
+            throw AzureError.client("Event code: \(code) is unsupported")
```

Comment:
I'm not sure I agree with the init throwing here. 
I think it might make more sense to have an `unknown` case. And have the consumer make the decision to throw an error in the case of `unknown`

Reply:
I added throw here because the caller which is signalling client cannot return a response to the users event handler if the code is unknown, right now the flow is:

1. Signalling client receives event payload
2. Checks code in payload
3. Based on code converts payload to one of the models in `chat.swift` ie 200 -> convert payload to a ChatMessageReceivedEvent
4. Returns that model to the event handler
5. If we don't recognize the code signaling client logs an error

I don't think the consumer is intended to be calling this but correct me if I'm wrong @glorialimicrosoft?